recipes-bsp/uefi/tegra-uefi-capsules_35.5.0.bb

```diff
@@ -2,54 +2,67 @@ DESCRIPTION = "Generate UEFI capsules for bup paylods"
 
 COMPATIBLE_MACHINE = "(tegra)"
-
-DEPENDS += "tegra-bup-payload"
+TEGRA_SIGNING_EXTRA_DEPS ??= ""
 
 GUID:tegra194 ?= "be3f5d68-7654-4ed2-838c-2a2faf901a78"
 GUID:tegra234 ?= "bf0d4599-20d4-414e-b2c5-3595b1cda402"
 
 do_compile() {
-    sign_uefi_capsules
+
+    if [ ! -z "${INITRAMFS_IMAGE}" ]; then
+        initramfs_symlink_name=${INITRAMFS_IMAGE}-${MACHINE}
+        # Generate BUP images
+        oe_make_bup_payload ${DEPLOY_DIR_IMAGE}/${initramfs_symlink_name}.cboot
+
+        # Generate UEFI capsules
+        generate_uefi_capsule ${WORKDIR}/bup-payload/bl_only_payload tegra-bl.cap
+        generate_uefi_capsule ${WORKDIR}/bup-payload/kernel_only_payload tegra-kernel.cap
+    fi
 }
 
-CAPSULE_DIR = "/opt/nvidia/UpdateCapsule"
+TEGRA_UEFI_CAPSULE_INSTALL_DIR ??= "/opt/nvidia/UpdateCapsule"
 
 do_install() {
-    install -d ${D}${CAPSULE_DIR}
+    install -d ${D}${TEGRA_UEFI_CAPSULE_INSTALL_DIR}
     if [ -e ${B}/tegra-bl.cap ]; then
-        install -m 0644 ${B}/tegra-bl.cap ${D}${CAPSULE_DIR}
+        install -m 0644 ${B}/tegra-bl.cap ${D}${TEGRA_UEFI_CAPSULE_INSTALL_DIR}
     fi
     if [ -e ${B}/tegra-kernel.cap ]; then
-        install -m 0644 ${B}/tegra-kernel.cap ${D}${CAPSULE_DIR}
+        install -m 0644 ${B}/tegra-kernel.cap ${D}${TEGRA_UEFI_CAPSULE_INSTALL_DIR}
     fi
 }
 
-FILES:${PN} += "${CAPSULE_DIR}"
+FILES:${PN} += "${TEGRA_UEFI_CAPSULE_INSTALL_DIR}"
 PACKAGE_ARCH = "${MACHINE_ARCH}"
 INSANE_SKIP:${PN} += "buildpaths"
 
 do_deploy() {
     install -d ${DEPLOYDIR}
     if [ -e ${B}/tegra-bl.cap ]; then
-	BL_NAME=${TNSPEC_MACHINE}-tegra-bl.cap
-	install -m 0644 ${B}/tegra-bl.cap ${DEPLOYDIR}/$BL_NAME
-	ln -s -r ${DEPLOYDIR}/$BL_NAME ${DEPLOYDIR}/tegra-bl.cap
+        BL_NAME=${TNSPEC_MACHINE}-tegra-bl.cap
+        install -m 0644 ${B}/tegra-bl.cap ${DEPLOYDIR}/$BL_NAME
+        ln -s -r ${DEPLOYDIR}/$BL_NAME ${DEPLOYDIR}/tegra-bl.cap
     fi
     if [ -e ${B}/tegra-kernel.cap ]; then
-	KERNEL_NAME=${TNSPEC_MACHINE}-tegra-kernel.cap
-	install -m 0644 ${B}/tegra-kernel.cap ${DEPLOYDIR}/$KERNEL_NAME
-	ln -s -r ${DEPLOYDIR}/$KERNEL_NAME ${DEPLOYDIR}/tegra-kernel.cap
+        KERNEL_NAME=${TNSPEC_MACHINE}-tegra-kernel.cap
+        install -m 0644 ${B}/tegra-kernel.cap ${DEPLOYDIR}/$KERNEL_NAME
+        ln -s -r ${DEPLOYDIR}/$KERNEL_NAME ${DEPLOYDIR}/tegra-kernel.cap
     fi
 }
 
 addtask deploy after do_install
-
+do_compile[depends] += "virtual/kernel:do_deploy tegra-flashtools-native:do_populate_sysroot dtc-native:do_populate_sysroot"
+do_compile[depends] += "python3-pyyaml-native:do_populate_sysroot"
+do_compile[depends] += "tegra-redundant-boot-rollback:do_populate_sysroot tegra-bootfiles:do_populate_sysroot"
+do_compile[depends] += "coreutils-native:do_populate_sysroot ${TEGRA_ESP_IMAGE}:do_image_complete virtual/secure-os:do_deploy"
+do_compile[depends] += "${TEGRA_SIGNING_EXTRA_DEPS}"
 do_compile[depends] += "${@bup_dependency(d)} ${TEGRA_UEFI_CAPSULE_SIGNING_EXTRA_DEPS}"
+addtask compile before do_build
```

Comment:
Is this needed?  do_compile should already be in the standard list of tasks for every recipe.